### PR TITLE
Refactor: Split location dataLoaders (ID and Term)

### LINKED ISSUE #688: Split locations dataloader into two dataloaders
We should have one dataloader for loading by id, and another by term.

### PR DESCRIPTION
**Pull request by aalises**
Now the location data loaders are separated based on the input (IDs or terms) as they call different API endpoints. In the future, we can add more Location dataLoaders based on other parameters (such as slugs) in a similar way.

 Closes #688

### CODE REVIEW

File: apps/core/components/searchForm/SearchForm.js [hunk 1]
```diff
@@ -103,8 +104,8 @@ class SearchForm extends React.Component<Props> {
         });
       }
       this.props.navigation.navigate(Routes.RESULTS, {
```
Comment (tbergq) [inline apps/core/components/searchForm/SearchForm.js]: Why is this change related to splitting data loaders?

File: apps/graphql/src/apps/location/dataloaders/LocationsByID.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        .join('&'),
+  );
+  return results.locations.map(location => sanitizeLocations([location]));
+};
+
+export default function locationsByIDLoader() {
+  return new OptimisticDataloader(
+    (
+      ids: $ReadOnlyArray<LocationInputID>,
+    ): Promise<Array<Location[] | Error>> => fetchLocations(ids),
```
Comment (tbergq) [inline apps/graphql/src/apps/location/dataloaders/LocationsByID.js]: Would it be possible that this returns `Promise<Array<Location | Error>>` instead. When querying for ID, you just expect to receive one result.

File: apps/graphql/src/apps/location/dataloaders/LocationsByID.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export type LocationInputID = {|
+  +code: string,
+|};
+
+const fetchLocations = async (params: $ReadOnlyArray<LocationInputID>) => {
+  const results = await fetch(
+    `/locations/id?limit=${params.length}&` +
+      params
+        .map((param: LocationInputID) => {
+          if (param.code !== undefined) {
```
Comment (tbergq) [inline apps/graphql/src/apps/location/dataloaders/LocationsByID.js]: Not sure we need this param.code !== undefined anymore

Comment (aalises) [reply apps/graphql/src/apps/location/dataloaders/LocationsByID.js]: We don't, on `RouteStop` the `code` can't be undefined

File: apps/graphql/src/apps/common/types/outputs/RouteStop.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ export default new GraphQLObjectType({
         if (code == null) {
           return null;
         }
-        const stops = await dataLoader.locations.load({ code });
+        const stops = await dataLoader.locations.byID.load({ code });
         if (!Array.isArray(stops) || stops.length < 1) {
           return null;
         }
```
Comment (tbergq) [inline apps/graphql/src/apps/common/types/outputs/RouteStop.js]: I see that the return type of the dataloader is changed, which is good, I see that this doesn't to give any flow-error which is bad. I want to get rid of this as well. 

```
 if (!Array.isArray(stops) || stops.length < 1) {
          return null;
  }
```
